In [1]:
import pandas as pd

In [2]:
# # Preparation for reading data from csv file on Google Colab
# # remove when reading the csv file locally from your computer
# import os
# from google.colab import drive
# drive.mount('/content/drive')
# os.chdir('/content/drive/MyDrive/HS/WIIM_DS/03_Discrete_Data_University_Applications')

In [3]:
d = pd.read_csv("studienzulassung.csv") # Load from file

In [4]:
d.head() # First Look at data

Fach Geschlecht  Angenommen
0       WI          M           0
1  IntMgmt          W           0
2  IntMgmt          W           0
3  IntMgmt          M           0
4       WI          M           0

In [14]:
pd.crosstab([d['Fach'], d['Geschlecht']],columns=d['Angenommen'], margins=True)

Angenommen               0    1  All
Fach       Geschlecht               
Informatik M            36   54   90
           W            18   27   45
IntMgmt    M            36    9   45
           W           108   27  135
WI         M            27   27   54
           W            18   18   36
All                    243  162  405

#### Exercise 1.1a)
We compute R(A | G) and compare the acceptance rate between men and women:

In [6]:
pd.crosstab(d['Geschlecht'],columns=d['Angenommen'], margins=True)

Angenommen    0    1  All
Geschlecht               
M            99   90  189
W           144   72  216
All         243  162  405

In [7]:
pd.crosstab(d['Geschlecht'],columns=d['Angenommen'], margins=True, normalize='all')

Angenommen         0         1       All
Geschlecht                              
M           0.244444  0.222222  0.466667
W           0.355556  0.177778  0.533333
All         0.600000  0.400000  1.000000

In [8]:
pd.crosstab(d['Geschlecht'],columns=d['Angenommen'], normalize='index', margins=True)

Angenommen         0         1
Geschlecht                    
M           0.523810  0.476190
W           0.666667  0.333333
All         0.600000  0.400000

#### Answer 1.1a:
Around 48% of men, but only about 33% auf women got accepted. The variables "gender" and "acceptance" are therefore statistically *dependent*.

##### OR:
R( A=1 | G='M') = 48%  !=  33% = R(A=1 | G='W') . A and G are therefore statistically *dependent*.

#### Exercise 1.1b)
We repeat the acceptance rate comparison from (a), but this time separately for the different subjects. To do this, we calculate R(A | G,F):

In [9]:
pd.crosstab([d['Fach'], d['Geschlecht']],columns=d['Angenommen'],margins=True)

Angenommen               0    1  All
Fach       Geschlecht               
Informatik M            36   54   90
           W            18   27   45
IntMgmt    M            36    9   45
           W           108   27  135
WI         M            27   27   54
           W            18   18   36
All                    243  162  405

In [10]:
pd.crosstab([d['Fach'], d['Geschlecht']],columns=d['Angenommen'], normalize='index')

Angenommen               0    1
Fach       Geschlecht          
Informatik M           0.4  0.6
           W           0.4  0.6
IntMgmt    M           0.8  0.2
           W           0.8  0.2
WI         M           0.5  0.5
           W           0.5  0.5

#### Answer 1.1b
For each of the subjects, the acceptance rate is the same for women and men. The different subject preferences between women and men therefore fully explain the dependency between gender and acceptance.   

Gender is useless for predicting the acceptance rate, given the subject

##### OR:
R(A=1 | Fach='Informatik', G='M') = 0.6 = R(A=1 | Fach='Informatik', G='W'),
R(A=1 | Fach='IntMgmt', G='M') = 0.2 = R(A=1 | Fach='IntMgmt', G='W'),
R(A=1 | Fach='WI', G='M') = 0.5 = R(A=1 | Fach='WI', G='W'),
Under the condition that the subject is known (or also "given the subject"), A and G are therefore statistically independent, i.e. F fully explains the relationship between G and A.

##### Plausibility check:
Of course this makes us curios whether the subject preferences between women and men are truly so different. We therefore calculate the subject distribution for women and men separately, i.e. R(F | G):

In [11]:
pd.crosstab(d['Geschlecht'],columns=d['Fach'], normalize='index')

Fach        Informatik   IntMgmt        WI
Geschlecht                                
M             0.476190  0.238095  0.285714
W             0.208333  0.625000  0.166667

In [12]:
pd.crosstab(d['Fach'],columns=d['Angenommen'], normalize='index')

Angenommen    0    1
Fach                
Informatik  0.4  0.6
IntMgmt     0.8  0.2
WI          0.5  0.5

Indeed, a much higher proportion of women (approx. 63%) apply for IntMgmt, where the acceptance rate is very low, than is the case for men (approx. 24%).


#### Answer 1.1c)
Before we read too much into the comparison of the above acceptance rates, let's check whether the absolute numbers of cases are large enough for the acceptance rates to be more than pure products of chance. This can be seen directly in the first table of question 1.1b: There are at least 36 cases for each subject-gender combination; in this respect, the numbers compared are not obviously useless. (For a quantitative statement, one could, for example, calculate statistical significance or a p-value.    

##### Answer 1.1 c-i)
F would suffice, since A is independent of G under the condition F. G therefore provides no additional information about A as soon as F is known.

##### Answer 1.1 c-ii)
There is no evidence of gender discrimination by the university, as the different acceptance rates are fully explained by the different subject choices. (However, this does not clarify how the gender differences in subject choice come about).

#### This was the solution to 1.1-d-ii). You can find the solution to 1.1-d-i as a PDF.